### DISCUSSION TITLE
Is there no access to plugins and other features without compiling yourself?

### DISCUSSION THREAD

**Discussion by rocket-pig** [massivemadness/Squircle-CE#157] [Help]
Hi,
First off thanks for an amazing Android CE, I've lived solely inside its beautiful functional walls for last few moons.

What im here to ask though - Are lowly APK dependent people such as myself unable to access things like hardware keyboard shortcut settings? Is there not a precompiled version where features like that are available?  Moving to line number without scrolling etc would be life-changing.

Sorry if dumb question or Im missing how to do it on the Play Store / FD version. 

Thanks again

**Comment by massivemadness** [comment]
Hi @rocket-pig, if I understand correctly, you want to change the keyboard shortcuts? It's not possible yet, but I have plans to make such feature in the next versions

**Comment by rocket-pig** [comment]
Nov 29, 2022 9:27:45 AM Dmitry Rubtsov ***@***.***>:

> Hi @rocket-pig[https://github.com/rocket-pig], if I understand correctly, you want to change the keyboard shortcuts? It's not possible yet, but I have plans to make such feature in the next versions
> 
> —
> Reply to this email directly, view it on GitHub[https://github.com/massivemadness/Squircle-CE/discussions/157#discussioncomment-4266530], or unsubscribe[https://github.com/notifications/unsubscribe-auth/AF5XFMOO5TI7L7YMC2GZTM3WKY4JBANCNFSM6AAAAAASOVPEUU].
> You are receiving this because you were mentioned.[Tracking image][https://github.com/notifications/beacon/AF5XFMKGVRZPWAHKE5YEYD3WKY4JBA5CNFSM6AAAAAASOVPEUWWGG33NNVSW45C7OR4XAZNRIRUXGY3VONZWS33OINXW23LFNZ2KUY3PNVWWK3TUL5UWJTQAIENCE.gif]
> 
Hi,
are you saying they're already available? Like I said, sorry if I'm missing the obvious.  I did not notice any keyboard shortcut mention in the App itself, and saw that there was a plugin architecture that included keyboard shortcuts..

Anyway if they work in the app already just aren't listed in the menus, sorry for the derp.

**Reply by massivemadness** [reply]
Yes they are already in the app [EditorFragment.kt#L298](https://github.com/massivemadness/Squircle-CE/blob/master/feature-editor/src/main/kotlin/com/blacksquircle/ui/feature/editor/ui/fragment/EditorFragment.kt#L298), but there is no option to change them

**Comment by rocket-pig** [comment]
Nov 29, 2022 9:44:07 AM Dmitry Rubtsov ***@***.***>:

> Yes they are already in the app EditorFragment.kt#L298[https://github.com/massivemadness/Squircle-CE/blob/master/feature-editor/src/main/kotlin/com/blacksquircle/ui/feature/editor/ui/fragment/EditorFragment.kt#L298], but there is no option to change them
> 
> —
> Reply to this email directly, view it on GitHub[https://github.com/massivemadness/Squircle-CE/discussions/157#discussioncomment-4266615], or unsubscribe[https://github.com/notifications/unsubscribe-auth/AF5XFMNKVCVEGX4WYKGIQGTWKY6GNANCNFSM6AAAAAASOVPEUU].
> You are receiving this because you were mentioned.[Tracking image][https://github.com/notifications/beacon/AF5XFMNQBQYQ4BCSCY6AKZ3WKY6GNA5CNFSM6AAAAAASOVPEUWWGG33NNVSW45C7OR4XAZNRIRUXGY3VONZWS33OINXW23LFNZ2KUY3PNVWWK3TUL5UWJTQAIENHO.gif]
> 
Awesome! Stoked. Nothing left to do but maybe suggest it's added to help, or app desciption on app stores? I compiled a list:

```
Paste: Ctrl-V
Cut: Ctrl-X
Copy: Ctrl-C
Undo: Ctrl-Z
Redo: Ctrl-Y
Save: Ctrl-S
Close: Ctrl-W
Find: Ctrl-F
Replace: Ctrl-R
Go to Line: Ctrl-G
Move Caret to Start of Line: Ctrl-Left
Move Caret to End of Line: Ctrl-Right
Move Caret to Previous Word: Alt-Left
Move Caret to Next Word: Alt-Right
Select Line: Alt-A
Settings: Alt-S
Tab: Tab
```

Thanks again
